src/coredump/coredump.c

```diff
@@ -1342,14 +1345,47 @@ static int gather_pid_metadata_from_argv(
                         t = buf;
                 }
 
+                if (i == META_ARGV_PIDFD) {
+                        if (!isempty(t)) {
+                                assert(kernel_fd < 0);
+
+                                kernel_fd = parse_fd(t);
+                                if (kernel_fd < 0)
+                                        return log_error_errno(kernel_fd, "Failed to parse PIDFD \"%s\": %m", t);
+
+                                r = pidref_set_pidfd(&context->pidref, kernel_fd);
+                                if (r < 0)
+                                        return log_error_errno(r,
+                                                                "Failed to initialize pidref from pidfd %d: %m",
+                                                                kernel_fd);
+
+                                r = pidref_verify(&context->pidref);
+                                if (r < 0)
+                                        return log_error_errno(r, "PIDFD validation failed: %m");
```

Comment:
this is pointless... piref_set_pidfd() immediately reads the pid of the pidfd, and stores it away, and pidref_verify() checks if the pid and the pidfd still match, or if the pid is zero now. but well, we queried it immediately before, didn#t do anything in between, hence there's no point in verifying the pidref.

verification of pidref is useful after you read some props off /proc/$PID/ and want to make sure the data actually belongs to the process in question before using it. but that's not done here